datafusion/core/tests/sqllogictests/test_files/timestamps.slt

```diff
@@ -261,6 +261,110 @@ SELECT INTERVAL '8' MONTH + '2000-01-01T00:00:00'::timestamp;
 ----
 2000-09-01T00:00:00
 
+# Interval columns are created with timestamp subtraction in subquery since they are not supported yet
```

Comment:
I think with https://github.com/apache/arrow-datafusion/pull/5792 we can now write better tests here -- specifically we can create interval constants.